docs/02-app/01-building-your-application/06-optimizing/13-memory-usage.mdx

```diff
@@ -0,0 +1,89 @@
+---
+title: Memory Usage
+description: Optimize memory used by your application in development and production.
+---
+
+As applications grow and become more feature rich, they can demand more resources when developing locally or creating production builds.
+
+Let's explore some strategies and techniques to optimize memory and address common memory issues in Next.js.
+
+## Reduce number of dependencies
+
+Applications with a large amount of dependencies will use more memory.
+
+The [Bundle Analyzer](/docs/app/building-your-application/optimizing/bundle-analyzer) can help you investigate large dependencies in your application that may be able to be removed to improve performance and memory usage.
+
+## Record a heap profile
+
+To look for memory issues, you can record a heap profile from Node.js and load it in Chrome DevTools to identify potential sources of memory leaks.
+
+In your terminal, pass the `--heap-prof` flag to Node.js when starting your Next.js build:
+
+```sh
+node --heap-prof next build
```

Comment:
Are you sure this is the right command? It's expecting a file

```bash
$ node --heap-prof next build
node:internal/modules/cjs/loader:1085
  throw err;
  ^

Error: Cannot find module 'A:\projects\46-fleetx\apps\vendor\next'
←[90m    at Module._resolveFilename (node:internal/modules/cjs/loader:1082:15)←
[39m
←[90m    at Module._load (node:internal/modules/cjs/loader:928:27)←[39m        
←[90m    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/
run_main:83:12)←[39m
←[90m    at node:internal/main/run_main_module:23:47←[39m {
  code: ←[32m'MODULE_NOT_FOUND'←[39m,
  requireStack: []
}

Node.js v20.0.0
```

Reply:
this should be `node --heap-prof ./node_modules/.bin/next build` depending on the package manager.

Reply:
Good catch, thank you. I sent https://github.com/vercel/next.js/pull/63982 to update